server/core/region.go

```diff
@@ -608,7 +608,51 @@ func (r *RegionsInfo) AddRegion(region *RegionInfo) []*RegionInfo {
 	}
 
 	r.regions.Put(region)
+	r.addRegionToSubTree(region)
+	return overlaps
+}
+
+// ReplaceOrAddRegion adds or replaces RegionInfo to regionTree and regionMap, also update leaders and followers by region peers
+func (r *RegionsInfo) ReplaceOrAddRegion(region *RegionInfo, overlaps []*RegionInfo, oldRegionItem *regionItem) {
+	// Add to tree and regions.
+	if !isEqualRegion(region, oldRegionItem) {
```

Comment:
even if the `region id` is the same, the leader may be changed, so maybe we need to remove the origin from the subtree.

Reply:
```
if origin.leader.GetId() != region.leader.GetId() {
				remove = true
			} else if !isEqualPeers(origin.GetVoters(), region.GetVoters()) {
				remove = true
			} else if !isEqualPeers(origin.GetLearners(), region.GetLearners()) {
				remove = true
			} else if !isEqualPeers(origin.GetPendingPeers(), region.GetPendingPeers()) {
				remove = true
			}
			if remove {
				r.removeRegionFromSubTree(origin)
			}
```
This code can realize the problem of subtree update after leader migration.

Reply:
The code is in `else` branch.